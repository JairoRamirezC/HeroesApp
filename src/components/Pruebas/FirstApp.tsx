import { useState } from "react";
import {CounterApp} from "./CounterApp";


const FirstApp = () => {
  const [counter, setCounter] = useState<number>(0);
  const [titles, setTitles] = useState<string>('');

  const handleIncrementCounter = () => {
    setCounter(prev => prev + 1);
  }
  
  const handleDecrementCounter = () => {
    if(counter === 0) return;
    setCounter(prev => prev - 1);
  }

  return (
    <>
      <p>{counter}</p>

      <button onClick={handleIncrementCounter}>Sumar +1</button>
      <button onClick={handleDecrementCounter}>Restar -1</button>
      <button onClick={() => setTitles('Nuevo nombre')}>Cambiar nombre</button>
      <CounterApp title={`${titles ? titles : 'React Memo'}`} subTitle="Si como no" name="Jairo" />
    </>
  )
}

export default FirstApp