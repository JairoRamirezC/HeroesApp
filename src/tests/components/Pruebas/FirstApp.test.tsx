import {render} from '@testing-library/react';
import FirstApp from '../../../components/Pruebas/FirstApp';

describe('Test in FirstApp', () => {

  test('FirstApp should be iqual to snapshot', () => {
    const { container } = render(<FirstApp />);
    expect(container).toMatchSnapshot();
  });

  test('FirstApp should render text "Cambiar nombre"', () => {
    const text = 'Cambiar nombre';
    const { getByText } = render(<FirstApp />);
    expect(getByText(text)).toBeTruthy();
  });
});