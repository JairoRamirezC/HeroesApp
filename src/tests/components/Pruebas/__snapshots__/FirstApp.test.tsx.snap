// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test in FirstApp FirstApp should be iqual to snapshot 1`] = `
<div>
  <p>
    0
  </p>
  <button>
    Sumar +1
  </button>
  <button>
    Restar -1
  </button>
  <button>
    Cambiar nombre
  </button>
  <div>
    <h1>
      React Memo
    </h1>
    <p>
      Si como no
    </p>
    <p>
      Jairo
    </p>
  </div>
</div>
`;

exports[`Test in FirstApp handleSum should return sum 1`] = `
<div>
  <p>
    0
  </p>
  <button>
    Sumar +1
  </button>
  <button>
    Restar -1
  </button>
  <button>
    Cambiar nombre
  </button>
  <div>
    <h1>
      React Memo
    </h1>
    <p>
      Si como no
    </p>
    <p>
      Jairo
    </p>
  </div>
</div>
`;
